l1-contracts/contracts/common/libraries/DataEncoding.sol

```diff
@@ -117,4 +117,32 @@ library DataEncoding {
             revert UnsupportedEncodingVersion();
         }
     }
+
+    function decodeTokenData(
+        bytes calldata _tokenData
+    ) internal pure returns (uint256 chainId, bytes memory name, bytes memory symbol, bytes memory decimals) {
+        bytes1 encodingVersion = _tokenData[0];
+        // kl todo check correct
+        if (encodingVersion == LEGACY_ENCODING_VERSION) {
```

Comment:
Will this work Stas?

Reply:
LGTM, though maybe a different constant variable should be introduced